### DISCUSSION TITLE
Crash in ReadBarcodes with ImageFormat::RGBA

### DISCUSSION THREAD

**Discussion by mattmcde78** [zxing-cpp/zxing-cpp#796] [Q&A] [answered]
I downloaded and compiled the master branch using MSVC 2022 on May 26.

Using the example code from [zxing-cpp](https://github.com/zxing-cpp/zxing-cpp/tree/master)/[example](https://github.com/zxing-cpp/zxing-cpp/tree/master/example/ZXingReader.cpp) , a crash occurs with the attached image.  The memory address pointed to by "src" in ZXing::SetupLumImageView is not valid.

ZXing::SetupLumImageView::__l14::<lambda>(const unsigned char * src) Line 115	C++
ZXing::ExtractLum<unsigned char <lambda>(const unsigned char *)>(const ZXing::ImageView & iv, ZXing::SetupLumImageView::__l14::unsigned char <lambda>(const unsigned char *) projection) Line 44	C++
ZXing::SetupLumImageView(ZXing::ImageView iv, ZXing::LumImage & lum, const ZXing::ReaderOptions & opts) Line 115	C++
ZXing::ReadBarcodes(const ZXing::ImageView & _iv, const ZXing::ReaderOptions & opts) Line 157	C++

I noted that stbi_load() returns that the image contains transparency information (4 channels).  If I change the image format from ImageFormat::RGBA to ImageFormat::RGB, there is no crash.

Can anyone else reproduce this crash?  Am I doing something wrong?

Thank you!

![Ref Guide 1 ](https://github.com/zxing-cpp/zxing-cpp/assets/128563382/12a39fb0-325a-42fb-ab5a-8e1c095e56f5)

**Comment by axxel** [comment]
I could not reproduce any memory access issue with ASAN on Linux. Since you closed the discussion you seem to have found the cause yourself? May I ask what it was?

**Reply by mattmcde78** [reply, marked as answer]
My apologies as it turned out to be my mistake.  When calling stbi_load, I was passing in a value of 3 for desired channels and not 0 as the example code does.  This caused it to return only RGB (24bits per pixel) data.  However, ImageFormat::RGBA (30bits per pixel) was then used to construct the ImageView which caused it to read past the buffer returned by stbi_load.
